File: DownloadLib/src/main/java/com/kronos/download/DownloadConfig.java
package com.kronos.download;

import android.os.Environment;
import android.text.TextUtils;

import okhttp3.OkHttpClient;

/**
 * Created by Leif Zhang on 2016/10/8.
 * Email leifzhanggithub@gmail.com
 */
public class DownloadConfig {
    private String downloadFolder;
    private OkHttpClient okHttpClient;
    private IDownloadDb downloadDb;

    public IDownloadDb getDownloadDb() {
        return downloadDb;
    }

    public void setDownloadDb(IDownloadDb downloadDb) {
        this.downloadDb = downloadDb;
    }

    public String getDownloadFolder() {
        return downloadFolder;
    }

    public OkHttpClient getOkHttpClient() {
        return okHttpClient;
    }

    public void setDownloadFolder(String downloadFolder) {
        this.downloadFolder = downloadFolder;
    }

    public void setOkHttpClient(OkHttpClient okHttpClient) {
        this.okHttpClient = okHttpClient;
    }

    public static class Builder {
        private String downloadFolder;
        private OkHttpClient okHttpClient;
        private IDownloadDb downloadDb;

        public Builder setDownloadDb(IDownloadDb downloadDb) {
            this.downloadDb = downloadDb;
            return this;
        }

        public Builder setDownloadFolder(String downloadFolder) {
            this.downloadFolder = downloadFolder;
            return this;
        }

        public Builder setOkHttpClient(OkHttpClient okHttpClient) {
            this.okHttpClient = okHttpClient;
            return this;
        }

        public DownloadConfig builder() {
            DownloadConfig downloadConfig = new DownloadConfig();
            if (TextUtils.isEmpty(downloadFolder)) {
                downloadFolder = Environment.getExternalStorageDirectory().getPath() + "/wallstreetcn/";
            }
            downloadConfig.setDownloadFolder(downloadFolder);
            if (okHttpClient == null) {
                okHttpClient = new OkHttpClient.Builder()
                        .build();
            }
            downloadConfig.setOkHttpClient(okHttpClient);
            if (downloadDb == null) {
                throw new NullPointerException();
            }
            downloadConfig.setDownloadDb(downloadDb);
            return downloadConfig;
        }
    }
}

File: DownloadLib/src/main/java/com/kronos/download/DownloadConfig.kt
package com.kronos.download

import android.os.Environment
import android.text.TextUtils

import okhttp3.OkHttpClient

/**
 * Created by Leif Zhang on 2016/10/8.
 * Email leifzhanggithub@gmail.com
 */
class DownloadConfig {
    var downloadFolder: String? = null
    var okHttpClient: OkHttpClient? = null
    var downloadDb: IDownloadDb? = null
    var settingConfig: DownloadSettingConfig? = null

    class Builder {
        private var downloadFolder: String? = null
        private var okHttpClient: OkHttpClient? = null
        private var downloadDb: IDownloadDb? = null
        private var settingConfig: DownloadSettingConfig? = null

        fun setDownloadDb(downloadDb: IDownloadDb): Builder {
            this.downloadDb = downloadDb
            return this
        }

        fun setDownloadFolder(downloadFolder: String): Builder {
            this.downloadFolder = downloadFolder
            return this
        }

        fun setOkHttpClient(okHttpClient: OkHttpClient): Builder {
            this.okHttpClient = okHttpClient
            return this
        }

        fun setSettingConfig(settingConfig: DownloadSettingConfig): Builder {
            this.settingConfig = settingConfig
            return this
        }

        fun builder(): DownloadConfig {
            val downloadConfig = DownloadConfig()
            if (TextUtils.isEmpty(downloadFolder)) {
                downloadFolder = Environment.getExternalStorageDirectory().path + "/wallstreetcn/"
            }
            downloadConfig.downloadFolder = downloadFolder
            if (okHttpClient == null) {
                okHttpClient = OkHttpClient.Builder()
                        .build()
            }
            downloadConfig.okHttpClient = okHttpClient
            if (downloadDb == null) {
                throw NullPointerException()
            }
            downloadConfig.downloadDb = downloadDb
            if (settingConfig == null) {
                settingConfig = DownloadSettingConfig()
            }
            downloadConfig.settingConfig = settingConfig
            return downloadConfig
        }
    }
}
